modules/core/arrow-test/src/main/kotlin/arrow/test/generators/Generators.kt

```diff
@@ -5,165 +5,173 @@ import arrow.core.*
 import arrow.data.*
 import arrow.typeclasses.Applicative
 import io.kotlintest.properties.Gen
+import io.kotlintest.properties.map
 import java.util.concurrent.TimeUnit
 
-fun <F, A> genApplicative(valueGen: Gen<A>, AP: Applicative<F>): Gen<Kind<F, A>> =
-  object : Gen<Kind<F, A>> {
-    override fun generate(): Kind<F, A> =
-      AP.just(valueGen.generate())
-  }
+fun <F, A> genApplicative(valueGen: Gen<A>, AP: Applicative<F>): Gen<Kind<F, A>> = create(
+        valueGen.random().map { AP.just(it) }
+)
 
-fun <A, B> genFunctionAToB(genB: Gen<B>): Gen<(A) -> B> =
-  object : Gen<(A) -> B> {
-    override fun generate(): (A) -> B {
-      val v = genB.generate()
-      return { _ -> v }
-    }
-  }
+fun <A, B> genFunctionAToB(genB: Gen<B>): Gen<(A) -> B> = create(
+        genB.random().map { { _: A -> it} }
+)
 
-fun <A> genFunctionAAToA(genA: Gen<A>): Gen<(A, A) -> A> =
-  object : Gen<(A, A) -> A> {
-    override fun generate(): (A, A) -> A {
-      val v = genA.generate()
-      return { _, _ -> v }
-    }
-  }
+fun <A> genFunctionAAToA(genA: Gen<A>): Gen<(A, A) -> A> = create(
+        genA.random().map { { _: A, _: A -> it} }
+)
 
 fun genThrowable(): Gen<Throwable> = object : Gen<Throwable> {
-  override fun generate(): Throwable =
-    Gen.oneOf(listOf(RuntimeException(), NoSuchElementException(), IllegalArgumentException())).generate()
+
+    override fun always() = listOf(RuntimeException(), NoSuchElementException(), IllegalArgumentException())
+
+    override fun random(): Sequence<Throwable> {
+        return Gen.from(always()).random()
```

Comment:
Can you please fix the `Gen.from` to a field so it keeps the seed across calls?

Reply:
The main problem with 3.0 is that random returns sequence that can be used only one time. So tests will crash when random will be invoked and used second time.

Reply:
Ooooooooooh. I see. It's different now. Yeah ignore this and the other I'll comment on.

Reply:
So I guess, I was wrong.